coalib/settings/ConfigurationGathering.py

```diff
@@ -105,8 +107,12 @@ def warn_nonexistent_targets(targets, sections, log_printer):
     :param sections:    The sections to search. (Dict.)
     :param log_printer: The log printer to warn to.
     """
+
+    sections_substrings = list(itertools.chain(
+        *[sections_name.split('.') for sections_name in sections]))
     for target in targets:
-        if target not in sections:
+        if (target not in sections_substrings and
+                target not in sections):
```

Comment:
nope ... If you have e.g. `python.all.somethig` and sections_substrings are `['python', 'all', 'somethig']